lib/main.js

```diff
@@ -10,139 +10,176 @@ import { CompositeDisposable } from 'atom';
+    // Default project to file directory if project path cannot be determined
     return dirname(filePath);
   }
   return atomProject;
-}
+};
 
-function filterWhitelistedErrors(stderr) {
+const filterWhitelistedErrors = (stderr) => {
   // Split the input and remove blank lines
   const lines = stderr.split(os().EOL).filter(line => !!line);
   const filteredLines = lines.filter(line =>
     // Only keep the line if it is not ignored
     !errorWhitelist.some(errorRegex => errorRegex.test(line)),
   );
   return filteredLines.join(os().EOL);
-}
-
-export function provideLinter() {
-  return {
-    name: 'Pylint',
-    grammarScopes: ['source.python', 'source.python.django'],
-    scope: 'file',
-    lintOnFly: true,
-    lint: (editor) => {
-      const filePath = editor.getPath();
-      const fileDir = dirname(filePath);
-      const fileText = editor.getText();
-      const projectDir = getProjectDir(filePath);
-      const cwd = workingDirectory.replace(/%f/g, fileDir).replace(/%p/g, projectDir);
-      const execPath = executable.replace(/%p/g, projectDir);
-      let format = messageFormat;
-      const patterns = {
-        '%m': 'msg',
-        '%i': 'msg_id',
-        '%s': 'symbol',
-      };
-      Object.keys(patterns).forEach((pattern) => {
-        format = format.replace(new RegExp(pattern, 'g'), `{${patterns[pattern]}}`);
-      });
-      const env = Object.create(process.env, {
-        PYTHONPATH: {
-          value: [
-            process.env.PYTHONPATH, fileDir, projectDir,
-            pythonPath.replace(/%f/g, fileDir).replace(/%p/g, projectDir),
-          ].filter(x => !!x).join(delimiter),
-          enumerable: true,
-        },
-        LANG: { value: 'en_US.UTF-8', enumerable: true },
-      });
-
-      const args = [
-        `--msg-template='{line},{column},{category},{msg_id}:${format}'`,
-        '--reports=n',
-        '--output-format=text',
-      ];
-      if (rcFile) {
-        args.push(`--rcfile=${rcFile.replace(/%p/g, projectDir).replace(/%f/g, fileDir)}`);
+};
+
+const fixPathString = (pathString, fileDir, projectDir) => {
+  const string = pathString;
+  const fRstring = string.replace(/%f/g, fileDir);
+  const pRstring = fRstring.replace(/%p/g, projectDir);
+  return pRstring;
+};
+
+const determineSeverity = (severity) => {
+  switch (severity) {
+    case 'error':
+    case 'warning':
+    case 'info':
+      return severity;
+    case 'convention':
+      return 'info';
+    default:
+      return 'warning';
+  }
+};
+
+export default {
+  activate() {
+    require('atom-package-deps').install('linter-pylint');
+
+    this.subscriptions = new CompositeDisposable();
+
+    this.subscriptions.add(atom.config.observe('linter-pylint.executablePath', (value) => {
+      this.executablePath = value;
+
+      // FIXME: Remove backwards compatibility in a future minor version
+      const oldPath = atom.config.get('linter-pylint.executable');
```

Comment:
Read the comment on the line directly above. This is backwards compatibility that will be removed in a month or two.